### ISSUE TITLE
DataLoader worker (pid xxx) is killed by signal when parameter nw > 0

### ISSUE THREAD

**Issue by TomHeaven** [jwyang/faster-rcnn.pytorch#376]
First I'd like to thank you for the great project.

I met a strange error when setting nw > 0 (e.g., 1, 2, 3...) training vgg16/resnet50 on pascal-voc 2007 dataset (or any other dataset I can find). For example, we I run
```shell
python trainval_net.py --dataset head --net res50 --bs 4 --nw 1 --cuda 
```
I get the following error:
```
RuntimeError: DataLoader worker (pid 893) is killed by signal: Unknown signal: 0. Details are lost due to multiprocessing. Rerunning with num_workers=0 may give better error trace.
```
When I set nw = 0, however, everything works smoothly.

I googled and somebody said that it may be caused by:
+ Insufficient memory. I'm pretty sure I have sufficient memory (32 GB).
+ Dataset may not prefer multiprocessing.

Any suggestions are welcome!

Running on:
+ pytorch 0.4.1
+ torchvision 0.2.1
+ Mac OS X 10.12.6
+ CUDA 9.0
+ CUDNN 7.0

**Comment by wangg12** [comment]
@TomHeaven Have you solved this problem?

**Comment by TomHeaven** [comment]
No. However, I didn't met the problem on another PC with Mac OS 10.13. So I guess there is something related to the operating system environment.

**Comment by zxcvb6958** [comment]
@TomHeaven Check whether your PC has enough share memory. This problem is solved when I double the memory(in my server).
